localstack/services/apigateway/models.py

```diff
@@ -82,7 +83,9 @@ def __init__(self):
 
         self.account.update(
             {
-                "cloudwatchRoleArn": arns.role_arn("api-gw-cw-role"),
+                "cloudwatchRoleArn": arns.role_arn(
+                    "api-gw-cw-role", DEFAULT_AWS_ACCOUNT_ID
+                ),  # FIXME: account ID must be of the current store
```

Comment:
This mimics the current behaviour where the account ID of the first APIGW request when the provider starts up. The effort involved to refactor APIGW codebase to fix this is substantial and out of the scope of this PR and will be addressed in the future.

Reply:
I see the problem here, the `_account_id` field of the store is not passed to the `__init__` called but manually attached later on to the store, so we can't access it here. Was there a specific reason it was done this way?

I don't think this mimics the current behaviour though:
```bash
$ awslocal apigateway get-account
{
    "cloudwatchRoleArn": "arn:aws:iam::000000000000:role/api-gw-cw-role",
    ...
}
$ AWS_ACCESS_KEY_ID=000000000002 awslocal apigateway get-account
{
    "cloudwatchRoleArn": "arn:aws:iam::000000000002:role/api-gw-cw-role",
    ...
}
```
This might create some issues, but I don't think we currently have the CloudWatch integration with APIGW, so it shouldn't be an IAM issue I hope.

Do you see how this could be achieved with the current architecture? Even a big a refactor in APIGW might not solve this if `ApiGatewayStore.account` is called in different places, we would need to manually update the account after the store instantiation. How can we address this issue?

Reply:
> Was there a specific reason it was done this way?

IIRC this was done to avoid large scale code refactoring, you see `RegionBackend`s, the precursor to `Store`, defined `__init__` functions. We wanted to introduce stores in a backward compatible manner.

> Do you see how this could be achieved with the current architecture?

Basically this ties back to the previous question, the most straightforward way is not to define business logic in the store definition. Initialiser functions and other member methods with business logic must not be defined as part of the store definition.

Reply:
I understand. I suppose we could get a quick fix once this is merged in `get_account` or something similar, to initialize the account value in the store if it doesn't exist instead. Maybe we could get something quick that wouldn't need a big refactor, I'll try to make something once this is merged. Thanks for the precisions!